mautrix_signal/matrix.py

```diff
@@ -100,6 +100,56 @@ async def handle_join(self, room_id: RoomID, user_id: UserID, event_id: EventID)
 
         await portal.handle_matrix_join(user)
 
+    async def handle_kick_ban(
+        self,
+        action: str,
+        room_id: RoomID,
+        user_id: UserID,
+        sender: UserID,
+        reason: str,
+        event_id: EventID,
+    ) -> None:
+        self.log.debug(f"{user_id} was {action} from {room_id} by {sender} for {reason}")
+        portal = await po.Portal.get_by_mxid(room_id)
+        if not portal:
+            return
+
+        if user_id == self.az.bot_mxid:
+            if portal.is_direct:
+                await portal.cleanup_and_delete()
+            return
```

Comment:
This part looks a bit weird.

DMs should never go to the other parts of the code and I'm not sure if kicking the bridge bot should unconditionally delete the DM (it's only required for e2ee, and the recipient is the main_intent in DM portals). The bot is required for group chats though, so unbridging when the bridge bot is kicked from a group would make sense.